cycles-ledger/src/storage.rs

```diff
@@ -1804,6 +1944,34 @@ fn reimburse(acc: Account, amount: u128, now: u64) -> Result<u64, ProcessTransac
     Ok(block_index)
 }
 
+// Reimburse an approval with a given amount
+fn reimburse_approval(
+    from: Account,
+    spender: Account,
+    amount: u128,
+    old_expires_at: Option<u64>,
+    now: u64,
+) -> Result<Nat, ApproveError> {
+    let (current_allowance, current_expiry) = allowance(&from, &spender, now);
+    let expires_at = if current_expiry > 0 {
+        Some(current_expiry)
+    } else {
+        old_expires_at
```

Comment:
The more I think about this the less sense it makes. If the current_expiry is 0 then you should not reimburse the approval because the approval has expired.

Reply:
That's a good observation!

Reply:
The `0` comes from an `unwrap_or_default`, so `0` means "Currently no approvals present". In that case we want to use the old one